pkg/controller/populators/util.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2023 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package populators
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	cc "kubevirt.io/containerized-data-importer/pkg/controller/common"
+)
+
+const (
+	primePvcPrefix = "prime"
+
+	// errCreatingPVCPrime provides a const to indicate we failed to create PVC prime for population
+	errCreatingPVCPrime = "ErrCreatingPVCPrime"
+	// createdPVCPrimeSuccessfully provides a const to indicate we created PVC prime for population
+	createdPVCPrimeSuccessfully = "CreatedPVCPrimeSuccessfully"
+
+	// AnnPopulatorKind annotation is added to the PVC' to specify the population kind, so it's later
+	// checked by the common populator watches.
+	AnnPopulatorKind = cc.AnnAPIGroup + "/storage.populator.kind"
+
+	// annMigratedTo annotation is added to a PVC and PV that is supposed to be
+	// dynamically provisioned/deleted by by its corresponding CSI driver
+	// through the CSIMigration feature flags. When this annotation is set the
+	// Kubernetes components will "stand-down" and the external-provisioner will
+	// act on the objects
+	annMigratedTo = "pv.kubernetes.io/migrated-to"
+
+	// importKind is used to specify that the PVC' is meant to be populated by the import flow
+	importKind = "import"
+	// uploadKind is used to specify that the PVC' is meant to be populated by the upload flow
+	uploadKind = "upload"
+)
+
+// IsPVCDataSourceRefKind returns if the PVC has DataSourceRef that
+// is equal to the given kind
+func IsPVCDataSourceRefKind(pvc *corev1.PersistentVolumeClaim, kind string) bool {
+	dataSourceRef := pvc.Spec.DataSourceRef
+	return dataSourceRef != nil && dataSourceRef.APIGroup != nil && *dataSourceRef.APIGroup == cc.AnnAPIGroup && dataSourceRef.Kind == kind
+}
+
+func isPVCPrimeDataSourceRefKind(pvc *corev1.PersistentVolumeClaim, kind string) bool {
+	owner := metav1.GetControllerOf(pvc)
+	if owner == nil || owner.Kind != "PersistentVolumeClaim" {
+		return false
+	}
+	populatorKind, _ := pvc.Annotations[AnnPopulatorKind]
+	switch populatorKind {
```

Comment:
I think its better to use the consts UploaSourceRef and ImportSourceRef instead of new consts and just return populatorKind == kind

Reply:
Agree, will change it